apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {levelTypeWithoutStatus} from '@cdo/apps/templates/progress/progressTypes';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import color from '@cdo/apps/util/color';
+import * as progressStyles from '@cdo/apps/templates/progress/progressStyles';
+import {getIconForLevel} from '@cdo/apps/templates/progress/progressHelpers';
+
+const styles = {
+  container: {
+    ...progressStyles.flex,
+    ...progressStyles.cellContent
+  },
+  icon: {
+    ...progressStyles.inlineBlock,
+    textAlign: 'center',
+    color: color.charcoal,
+    fontSize: 20
+  }
+};
+
+export default class ProgressTableLevelIcon extends React.Component {
+  static propTypes = {
+    levels: PropTypes.arrayOf(levelTypeWithoutStatus)
+  };
+
+  render() {
+    return (
+      <span style={styles.container}>
+        {this.props.levels.map(level => {
```

Comment:
This whole return statement is a little difficult to read.  Maybe try extracting the anonymous function from L30-L50?  If it's still difficult to read, then maybe also extract L32-40 and L42-48?  (Extracting these parts gives you a place to write some comments which I think might be helpful, especially that the sublevels bit is to add space to get icons to line up (right?))

Reply:
I did a little refactor, let me know if you find it more readable!